src/Rules/Functions/ImplodeFunctionRule.php

```diff
@@ -79,4 +88,21 @@ static function (Type $type): bool {
 		];
 	}
 
+	private function checkContainsErrorTypeAfterConversion(Type $type): bool
```

Comment:
Should I extract this to separate PR? Unions were not handled correctly

Reply:
Looks unrelated to short circuiting. If you found a bug in this code, please submit it as a separate PR and remove the change from this PR. Thanks.